src/lib/core-extension-service.ts

```diff
@@ -75,6 +79,13 @@ export class CoreExtensionService {
 		}
 	}
 
+	private static initializeWorkspaceContext(workspaceContext: WorkspaceContext | undefined) {
+		if (!workspaceContext) {
+			throw new Error('***workspace context not found***');
+		}
+		CoreExtensionService.workspaceContext = workspaceContext.getInstance(false);
+	}
```

Comment:
This will always error if someone installs code analyzer without the extension pack it seems (because the coreExtensionApi can't be pulled in since it is pulled on line 55 using the CORE_EXTENSION_ID.

At least with the telemetry service, we don't error out of it doesn't exist. Shouldn't we do something similar here?

Reply:
ApexGuru integration will not work if someone does have the core extension. That's because of all the auth dependencies. So, unlike the telemetry service, we do want to error out. We need better messaging around this when we release this. I hope us being part of the extension pack is a good thing, so not bad overall. Like some of the extension, we could put a note that instead of installing the separate extension, encourage users to install the pack.

Reply:
My point is... we shouldn't require it. Right now.. the plugin (once the boolean is flipped to true) is completely unusable because of this error. initializeWorkspaceContext is called always and if they don't have the extension pack, then an error is thrown it seems.

Instead, why not just disable the apexGuru functionality if this can't be found (just like how telemetry services are disabled if it can't be found).